azure-functions/acbs-function/mappings/facility/facility-loan-amend.js

```diff
@@ -0,0 +1,46 @@
+/*
+"amount":   ukefExposure
+*/
+
+const CONSTANTS = require('../../constants');
+const { formatDate } = require('../../helpers/date');
+const helpers = require('./helpers');
+
+const facilityLoanAmend = (amendments, facilityMasterRecord) => {
+  try {
+    let record = {
+      portfolioIdentifier: CONSTANTS.FACILITY.PORTFOLIO.E1,
+    };
+    const { amendment } = amendments;
+
+    if (amendment) {
+      const { amount, coverEndDate } = amendment;
+
+      // UKEF Exposure
+      if (amount) {
+        // Only BSS (Bond) facility types are subjected to loan amount amendment.
+        if (facilityMasterRecord.productTypeId === CONSTANTS.FACILITY.FACILITY_TYPE_CODE.BSS) {
+          record = {
+            ...record,
+            amount: helpers.getMaximumLiability(amendments),
+          };
+        }
+      }
+
+      // Cover end date
+      if (coverEndDate) {
+        record = {
+          ...record,
+          expiryDate: formatDate(coverEndDate),
+        };
+      }
+    }
+
+    return record;
+  } catch (e) {
+    console.error('Unable to map facility loan record: ', { e });
+    return e;
```

Comment:
in other files we've used `throw new Error()`. Do we need to do the same here?

Reply:
Nope, need to throw error for durable activity functions so they can be stored in Azure storage.
In this case (mapping function) if an error is thrown it will be caught through-out the call stack and displayed on console only.